Tools/autotest/arducopter.py

```diff
@@ -965,6 +965,30 @@ def setup_rc(mavproxy):
     mavproxy.send('rc 3 1000\n')
 
 
+def fly_throw_mode(mavproxy, mav):
+        # test boomerang mode:
+        progress("Throwing vehicle away")
+        set_parameter(mavproxy, "THROW_NEXTMODE", 6)
+        set_parameter(mavproxy, "SIM_SHOVE_Z", -30)
+        set_parameter(mavproxy, "SIM_SHOVE_X", -20)
+        mavproxy.send('mode throw\n')
+        wait_mode(mav, 'THROW')
+        mavproxy.send('arm throttle\n')
+        mav.motors_armed_wait()
+        set_parameter(mavproxy, "SIM_SHOVE_TIME", 500)
```

Comment:
This just showed an issue with MAVProxy (or at least our autotest interaction with it via command-line): SITL is receiving and accepting the 500 value for the parameter and certainly it is sending it back. But MAVProxy will only print the parameter value if a `fetch` command was issued - so `set_parameter` does a `fetch` after the `set`, but apparently that's not fast enough and MAVProxy is always getting the parameter with value 0 back (and so it continues trying to set it even after the throw happened: https://travis-ci.org/ArduPilot/ardupilot/jobs/307843138#L3366-L3372)

My suggestion (and I think it is a reasonable one) would be that MAVProxy adds the parameter to `fetch_one` variable when the `set` command is called - this way when setting a variable in MAVProxy it will show what it got back.
I should add that the `set` command calls `mavset` (from pymavlink _mavparm.py_) that already checks if it gets back the same parameter in a PARAM_VALUE message (it doesn't check that the set value was the one received though).

Reply:
@OXINARF I'm not sure that we're not fetching the parameter back after the test has entirely finished.  We reset that parameter to 0 at the end of the test.

Reply:
Looking at the log it doesn't look like it. Don't forget that the parameter is reset to 0 after the launch time has passed.